pkg/server/ca/manager/manager.go

```diff
@@ -712,6 +811,35 @@ func (m *Manager) appendBundle(ctx context.Context, caChain []*x509.Certificate,
 	return res, nil
 }
 
+// prepareUntaintedX509CA prepare nextX509CA until it is no longer tainted
+func (m *Manager) prepareUntaintedX509CA(ctx context.Context, taintedAuthorities []*x509.Certificate) error {
+	// Prepare a new X.509 authority when next is old or it is signed by a tainted key
+	if ok := m.shouldPrepareX509CA(taintedAuthorities); !ok {
+		return nil
+	}
+
+	err := m.PrepareX509CA(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to prepare x509 authority: %w", err)
+	}
+
+	return nil
+}
+
+func (m *Manager) shouldPrepareX509CA(taintedAuthorities []*x509.Certificate) bool {
+	slot := m.nextX509CA
+	switch {
+	case slot.IsEmpty():
+		return true
+	case slot.Status() == journal.Status_PREPARED:
+		isTainted := isX509AuthorityTainted(slot.x509CA, taintedAuthorities)
+		m.c.Log.Debug("Next authority is tainted, prepare new X.509 authority")
```

Comment:
I think that logging this at Info level is more appropriate.
```suggestion
		m.c.Log.Info("Next authority is tainted, prepare new X.509 authority")
```